client/state/admin-menu/actions/index.js

```diff
@@ -16,3 +20,36 @@ export const receiveAdminMenu = function receiveAdminMenu( siteId, menu ) {
 		menu,
 	};
 };
+
+export const updateAdminMenuAfterMarketplaceInstallation =
+	( siteId, productSlug ) => ( dispatch, getState ) => {
+		const { transfer } = getLatestAtomicTransfer( getState(), siteId );
+		const pluginOnSite = getPluginOnSite( getState(), siteId, productSlug );
+
+		// Wait for the Atomic transfer to be completed...
+		if ( transfer?.status !== 'completed' ) {
+			// Check again after 2s.
+			dispatch( requestLatestAtomicTransfer( siteId ) );
+			setTimeout(
+				() => dispatch( updateAdminMenuAfterMarketplaceInstallation( siteId, productSlug ) ),
+				2000
+			);
+			return;
+		}
+
+		// ...And for the plugin to be activated.
+		if ( ! pluginOnSite ) {
+			// Check again after 2s.
+			dispatch( fetchSitePlugins( siteId ) );
+			setTimeout(
+				() => dispatch( updateAdminMenuAfterMarketplaceInstallation( siteId, productSlug ) ),
+				2000
+			);
+			return;
+		}
+
+		// Wait for another 2s before updating the menu to allow the plugin to finish any potential setup.
+		setTimeout( () => {
+			dispatch( requestAdminMenu( siteId ) );
+		}, 2000 );
```

Comment:
I'm wondering if these timeouts make the intention of this change moot, where we're waiting so long to update the menu that users have already clicked away by the time this request resolves?

Reply:
Yeah, I was being extra cautious for no apparent reason. I've removed the final delay in https://github.com/Automattic/wp-calypso/pull/65293/commits/4fbf9e08b32933ab0af064e6767ef63cee05383f. We can always add it if we're certain that this can happen.